rancher2/resource_rancher2_machine_config_v2.go

```diff
@@ -46,9 +46,20 @@ func resourceRancher2MachineConfigV2Create(d *schema.ResourceData, meta interfac
 		Delay:      1 * time.Second,
 		MinTimeout: 3 * time.Second,
 	}
-	_, waitErr := stateConf.WaitForState()
-	if waitErr != nil {
-		return fmt.Errorf("[ERROR] waiting for machine config (%s) to be active: %s", newObj.ID, waitErr)
+
+	// Terraform may finish creating the machine config v2 before rancher has set get
+	// permissions on the resource. Retry a few times to see if the state will return
+	// without error.
+	for i := 0; i < 4; i++ {
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			if IsNotAccessibleByID(err) {
+				log.Printf("[INFO] Retrying state for Machine Config V2 %s", newObj.ID)
+				continue
+			}
+			return fmt.Errorf("[ERROR] waiting for machine config (%s) to be active: %s", newObj.ID, err)
+		}
+		return flattenMachineConfigV2(d, newObj)
```

Comment:
From what I've seen the machine config v2 state reconciles after 1 try in terraform, so doing 4 is just for good measure but yeah, ideally we'd want to return an err saying if the machine config _still_ isn't accessible then return an error from the create function.

Reply:
Yeah I think we should do that, because we could potentially have a problem with the controller that is supposed to give the user access to that machine config, and in that case we would want to return the error even if we can never error after 4 tries in the happy path.

Reply:
Ah ok makes sense. I'll add an extra check

Reply:
Added! Thanks for me catching me on this one @jakefhyde